interpreter/interpretable.go

```diff
@@ -677,7 +677,15 @@ func (m *evalMap) Eval(ctx Activation) ref.Val {
 }
 
 func (m *evalMap) InitVals() []Interpretable {
-	return append(m.keys, m.vals...)
+	if len(m.keys) != len(m.vals) {
+		return nil
+	}
+	var result []Interpretable
```

Comment:
You can also size this to be the same length as `result := make([]Interpretable, len(m.keys) + len(m.vals))` and then assign key and values to indices, but this is mostly cosmetic though it might slightly improve performance.

Reply:
Can't hurt.